inference-engine/thirdparty/clDNN/src/include/primitive_inst.h

```diff
@@ -21,11 +21,19 @@
 
 namespace cldnn {
 
+// checks if any user in a list is a cpu primitive
+bool is_any_user_cpu(const std::list<const program_node*>& users);
+
 class primitive_inst;
 
 template <class PType>
 class typed_primitive_inst;
 
+struct internal_buffer_info {
+    data_types dtype;
+    std::vector<size_t> sizes;
```

Comment:
Maybe it should be a vector of `cldnn::layout` as this structure is converted to layouts anyway.

Reply:
Currently internal buffers are allocated as 1D layout {1, 1, 1, x}, but should we need layout info here?

Reply:
No, you can still keep it as {1,1,1,x} for now, but it's specific for ocl primitive impls and should be specified there. Maybe for other implementation types (e.g. onednn) we'll need to allocate memory with proper layout, and with proposed change it will be possible and quite easy.

Reply:
Okay, I understood.

Reply:
I removed this and used std::vector<layout>